controllers/chat/moveToAnon.js

```diff
@@ -30,6 +31,11 @@ const moveToAnon = async (req, res) => {
     user_id: targetUserId
   });
 
+  const [emoji, color] = await Promise.all([
+    BetterSocialConstantListUtils.getRandomEmoji(),
+    BetterSocialConstantListUtils.getRandomColor()
+  ]);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `Promise.all()` function is used here to fetch a random emoji and color concurrently. However, if one of these promises fails, the other will also be rejected immediately, which might not be the desired behavior. Consider handling each promise separately to allow for more granular error handling.

```diff
- const [emoji, color] = await Promise.all([
-   BetterSocialConstantListUtils.getRandomEmoji(),
-   BetterSocialConstantListUtils.getRandomColor()
- ]);
+ let emoji, color;
+ try {
+   emoji = await BetterSocialConstantListUtils.getRandomEmoji();
+ } catch (error) {
+   console.error('Failed to get random emoji:', error);
+   return;
+ }
+ try {
+   color = await BetterSocialConstantListUtils.getRandomColor();
+ } catch (error) {
+   console.error('Failed to get random color:', error);
+   return;
+ }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->